compiler/ballerina-lang/src/main/resources/grammar/BallerinaParser.g4

```diff
@@ -122,7 +122,7 @@ constantDefinition
 
 globalVariableDefinition
     :   PUBLIC? LISTENER? typeName Identifier (ASSIGN expression)? SEMICOLON
-    |   PUBLIC? FINAL (typeName | VAR) Identifier ASSIGN expression SEMICOLON
+    |   PUBLIC? FINAL? (typeName | VAR) Identifier ASSIGN expression SEMICOLON
```

Comment:
We can merge this line with line 124. Right now we don't support var with listeners. But we should be able to support that. Could you please create an issue as an improvement, if we can't fix that with this PR.

Reply:
Issue https://github.com/ballerina-platform/ballerina-lang/issues/13007 created to track this.